libobs/CMakeLists.txt

```diff
@@ -219,6 +219,16 @@ target_sources(
     graphics/vec4.h
 )
 
+if(CMAKE_C_COMPILER_ID STREQUAL "MSVC")
+  set_source_files_properties(obs-data.c PROPERTIES COMPILE_OPTIONS "/wd4996")
+elseif(
+  CMAKE_C_COMPILER_ID STREQUAL "GNU"
+  OR CMAKE_C_COMPILER_ID STREQUAL "Clang"
+  OR CMAKE_C_COMPILER_ID STREQUAL "AppleClang"
+)
+  set_source_files_properties(obs-data.c PROPERTIES COMPILE_OPTIONS "-Wno-deprecated-declarations")
+endif()
```

Comment:
If we're doing this here, we might need a comment explaining why and that this _should_ be temporary.

Reply:
As a follow-up we might want to add a common helper function that takes a list of sources files for a target and adds that for each, something like `target_silence_deprecations(<target> DISABLE <files> DOWNGRADE <files>)`.

Not needed for this PR, but given that we want to deprecate and obsolete things moving forward, I'd guess we might use that more often as well.

Reply:
Comment added. Hopefully that and the blame are enough for us to remember to remove this when we remove the autoselect functions.